torch/csrc/jit/passes/python_print.cpp

```diff
@@ -504,19 +519,75 @@ class PythonPrintPass {
           }
           stmt << "]";
         } else if(v.isBoolList()) {
-          printMaybeAnnoatatedConstantList(stmt, "bool", v.toBoolListRef().size(), v);
+          printMaybeAnnotatedConstantList(stmt, "bool", v.toBoolListRef().size(), v);
         } else if(v.isIntList()) {
-          printMaybeAnnoatatedConstantList(stmt, "int", v.toIntListRef().size(), v);
+          printMaybeAnnotatedConstantList(stmt, "int", v.toIntListRef().size(), v);
         } else if(v.isDoubleList()) {
-          printMaybeAnnoatatedConstantList(stmt, "float", v.toDoubleListRef().size(), v);
+          printMaybeAnnotatedConstantList(stmt, "float", v.toDoubleListRef().size(), v);
         } else {
           stmt << v;
         }
       } break;
-      case prim::None:
+      case prim::None: {
+        if (node->output()->type()->isSubtypeOf(NoneType::get())) {
+          stmt << "None";
+          break;
+        }
+        // XXX - when None has an Optional[T] type, we must ensure that type
```

Comment:
discussed in person: because of the difficulty of recovering the T in an `unpack_optional(None)`,  we need to make `prim::None` have the ability to have type `optional[T]` for some `T`.